src-tauri/src/storage/karma_repository.rs

```diff
@@ -0,0 +1,49 @@
+use async_trait::async_trait;
+
+use crate::model::karma::KarmaPoint;
+use crate::storage::db::{DbManager, DbManagerError};
+
+#[async_trait]
+pub trait KarmaRepository {
+    async fn insert_karma(&self, karma: KarmaPoint) -> Result<KarmaPoint, DbManagerError>;
+}
+
+#[async_trait]
+impl KarmaRepository for DbManager {
+    async fn insert_karma(&self, karma: KarmaPoint) -> Result<KarmaPoint, DbManagerError> {
+        let _query_result = sqlx::query("INSERT INTO karma(purpose_type, name) VALUES(?, ?);")
```

Comment:
An alternative that will increase readability:
1. Import query function and query::Query trait
2. use query! macro i.e. :
use sqlx::query;
use sqlx::query::Query;

let _query_result = query!(
    "INSERT INTO karma(purpose_type, name) VALUES(?, ?)",
    karma.get_purpose() as i32,
    karma.get_name()
)
.execute...;
Adding update, delete and other ops on karma will keep the code clearer, shorter and more readable.

Reply:
I will explore this option. It's not ideal for sqlite because the db is just a file and the query! needs the DATABASE_URL env var to be set in order to compile